lib/auth.guard.ts

```diff
@@ -105,6 +105,7 @@ const createPassportContext =
           request.authInfo = info;
           return resolve(callback(err, user, info, status));
         } catch (err) {
+          new Logger('AuthGuard').error(`${err.response.message} (${info})`);
```

Comment:
Instead of inline instantiation, could we make a class member for the logger and call `this.logger` inside of the `handleRequest `(that's where the code flows to whether pass or fail and we could log out before the `throw new UnauthorizedException()`)

Reply:
Yeah, just not a huge fan of inline `new Class().doTheThing()`. Feels harder to read in my opinion

Reply:
Modification done, tell me what you think !